src/browser/api_protocol/transport_strategy/elipc_strategy.ts

```diff
@@ -151,10 +151,30 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
         };
     }
 
-    protected ackDecorator(e: any, messageId: number): AckFunc {
+    private addBreadcrumb(message: any,  messageId: number, action: string, name: string): void {
+        if (message && message.breadcrumbs && message.breadcrumbs.length) {
```

Comment:
This conditional should always be true if the function is getting called, right? If so, it's probably worth putting an else block that prints an error saying the message was invalid (and the message itself).

Reply:
This is a high-transactional call when enabled. The debug.log could become flooded with the error message. I can adjust the implementation to add the array when it is not present.

Reply:
That sounds good - maybe add the array if not present, and log?

Reply:
I'm advocating to avoid logging, this would be called on *every* API message when enabled

Reply:
I'm thinking that because every call of `addBreadcrumb` (at the moment) is conditional on `usingBreadcrumbs` being true, we should always expect the `breadcrumbs` array to be present, so if it's not it's something we'd want to note by logging. If we expect to call `addBreadcrumb` without that conditional, then it wouldn't make as much sense to log.